search5/src/main/java/org/hibernate/performance/search/model/application/SearchProgrammaticMapping.java

```diff
@@ -0,0 +1,149 @@
+		questions.indexEmbedded().depth( 1 );
+		questions.containedIn();
+
+		// Question
+		IndexedMapping question = mapping.entity( Question.class ).indexed();
+		question.property( "id", ElementType.FIELD ).documentId();
+		question.property( "text", ElementType.FIELD ).field();
+
+		PropertyMapping questionnaire = question.property( "questionnaire", ElementType.FIELD );
+		questionnaire.indexEmbedded().depth( 1 );
+		questionnaire.containedIn();
+
+		// OpenQuestion
+		mapping.entity( OpenQuestion.class ).indexed();
+
+		// ClosedQuestion
+		IndexedMapping closedQuestion = mapping.entity( ClosedQuestion.class ).indexed();
+		closedQuestion.property( "weight", ElementType.FIELD ).field().numericField();
+
+		// QuestionnaireInstance
+		IndexedMapping questionnaireInstance = mapping.entity( QuestionnaireInstance.class ).indexed();
+		questionnaireInstance
+				.property( "id", ElementType.FIELD ).documentId()
+				.property( "uniqueCode", ElementType.FIELD ).field().analyze( Analyze.NO )
+				.property( "definition", ElementType.FIELD ).indexEmbedded()
+				.property( "approval", ElementType.FIELD ).indexEmbedded()
+				.property( "subject", ElementType.FIELD ).indexEmbedded()
+				.property( "notes", ElementType.FIELD ).field();
+
+		PropertyMapping closedAnswers = questionnaireInstance.property( "closedAnswers", ElementType.FIELD );
+		closedAnswers.indexEmbedded().depth( 1 );
+		closedAnswers.containedIn();
+
+		PropertyMapping openAnswers = questionnaireInstance.property( "openAnswers", ElementType.FIELD );
+		openAnswers.indexEmbedded().depth( 1 );
+		openAnswers.containedIn();
+
+		// OpenAnswer
+		IndexedMapping openAnswer = mapping.entity( OpenAnswer.class ).indexed();
+		openAnswer.property( "id", ElementType.FIELD ).documentId();
+		openAnswer.property( "text", ElementType.FIELD ).field();
+		openAnswer.property( "question", ElementType.FIELD ).indexEmbedded().depth( 2 );
+
+		PropertyMapping questionnairePropertyOA = openAnswer.property( "questionnaire", ElementType.METHOD );
+		questionnairePropertyOA.indexEmbedded().depth( 1 );
+		questionnairePropertyOA.containedIn();
+
+		// ClosedAnswer
+		IndexedMapping closedAnswer = mapping.entity( ClosedAnswer.class ).indexed();
+		closedAnswer.property( "id", ElementType.FIELD ).documentId();
+		closedAnswer.property( "choice", ElementType.FIELD ).field().numericField();
+		closedAnswer.property( "question", ElementType.FIELD ).indexEmbedded().depth( 2 );
+
+		PropertyMapping questionnairePropertyCA = closedAnswer.property( "questionnaire", ElementType.METHOD );
+		questionnairePropertyCA.indexEmbedded().depth( 1 );
+		questionnairePropertyCA.containedIn();
+
+		// PerformanceSummary
+		IndexedMapping performanceSummary = mapping.entity( PerformanceSummary.class ).indexed();
+		performanceSummary
+				.property( "id", ElementType.FIELD ).documentId()
+				.property( "employee", ElementType.FIELD ).indexEmbedded().depth( 3 )
```

Comment:
If you do this, I think there should be a `containedIn()` from `Employee` to `PerformanceSummary`. Employees are expected to have their data change from time to time, and it wouldn't cost much to reindex their perf summaries when that happens.

Reply:
Also, maybe you could use `includePaths` here? You probably don't need to index-embed *everything*, and I know for a fact that when using `includePaths`, Search 6 will be smarter and skip reindexing in some cases.